models/contacts.js

```diff
@@ -1,19 +1,103 @@
-// const fs = require('fs/promises')
+const fs = require("fs").promises;
+const { v4: uuidv4 } = require("uuid");
+const Joi = require("joi");
 
-const listContacts = async () => {}
+const saveContact = async (c) =>
+  await fs.writeFile("models/contacts.json", JSON.stringify(c));
 
-const getContactById = async (contactId) => {}
+const getContacts = async () => {
+  const contacts = fs.readFile("models/contacts.json");
+  return JSON.parse(await contacts);
+};
 
-const removeContact = async (contactId) => {}
+const listContacts = async () => {
+  try {
+    return await getContacts();
+  } catch (error) {
+    console.log(error);
+  }
+};
 
-const addContact = async (body) => {}
+const getContactById = async (contactId) => {
+  try {
+    const contacts = await getContacts();
 
-const updateContact = async (contactId, body) => {}
+    return contacts.find((c) => c.id === contactId);
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+const removeContact = async (contactId) => {
+  try {
+    const contacts = await getContacts();
+    const findContactToDelete = contacts.find((c) => c.id === contactId);
+
+    if (findContactToDelete) {
+      const contactsListAfterRemove = contacts.filter(
+        (contact) => contact.id !== contactId
+      );
+      await saveContact(contactsListAfterRemove);
+
+      return true;
+    } else {
+      return false;
+    }
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+const addContact = async (body) => {
+  try {
+    const { name, email, phone } = body;
+    const addedContact = {
+      id: uuidv4(),
+      name,
+      email,
+      phone,
+    };
+
+    const contacts = await getContacts();
+    contacts.push(addedContact);
+    await saveContact(contacts);
+
+    return addedContact;
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+const updateContact = async (contactId, body) => {
+  try {
+    const { name, email, phone } = body;
+    const editContact = {
+      name,
+      email,
+      phone,
+    };
+
+    const contacts = await getContacts();
+    const index = contacts.findIndex((c) => c.id === contactId);
```

Comment:
please dont use short names. Instead of `c` please use `contact`. Its easier to read and understand what's going on in code.

Reply:
done